pipes/rules/assembly.rules

```diff
@@ -49,12 +50,36 @@ rule assemble_trinity:
         UGER        = config.get('UGER_queues', {}).get('short', '-l h_rt 04:00:00'),
         n_reads     = str(config["trinity_n_reads"]),
         logid       = "{sample}",
-        subsamp_bam = config["tmp_dir"]+'/'+config["subdirs"]["assembly"]+'/{sample}.subsamp.bam'
     run:
         makedirs(expand("{dir}/{subdir}",
             dir=[config["data_dir"],config["tmp_dir"]],
             subdir=config["subdirs"]["assembly"]))
-        shell("{config[bin_dir]}/assembly.py assemble_trinity {input.bam} {input.clipDb} {output.fasta} --n_reads={params.n_reads} --outReads {params.subsamp_bam} --threads {resources.threads} --JVMmemory 5g")
+        shell("{config[bin_dir]}/assembly.py assemble_trinity {input.bam} {input.clipDb} {output.fasta} --n_reads={params.n_reads} --outReads {output.subsamp_bam} --threads {resources.threads} --JVMmemory 5g")
+
+rule assemble_spades:
+    ''' This step runs the SPAdes assembler.
+    '''
+    input:  taxfilt_reads=config["data_dir"]+'/'+config["subdirs"]["per_sample"]+'/{sample}.taxfilt.bam',
+            subsamp_bam=config["tmp_dir"]+'/'+config["subdirs"]["assembly"]+'/{sample}.subsamp.bam',
+            contigs_trinity=config["tmp_dir"] +'/'+config["subdirs"]["assembly"]+'/{sample}.assembly1-trinity.fasta'
+    output: contigs_spades=config["tmp_dir"] +'/'+config["subdirs"]["assembly"]+'/{sample}.assembly1-spades.fasta',
+            contigs_trinity_spades=config["tmp_dir"] +'/'+config["subdirs"]["assembly"]+'/{sample}.assembly1-trinity-spades.fasta'
+    resources: 
+            mem=7,
+            cores=int(config.get("number_of_threads", 1))
+    params: LSF=config.get('LSF_queues', {}).get('short', '-W 4:00'),
+            UGER=config.get('UGER_queues', {}).get('short', '-q short'),
+            n_reads=str(config["trinity_n_reads"]),
+            logid="{sample}",
+            clipDb=config["trim_clip_db"],
+            subsamp_bam=config["tmp_dir"]+'/'+config["subdirs"]["assembly"]+'/{sample}.subsamp.bam',
+            numThreads=str(config.get("number_of_threads", 1))
+    run:
+            makedirs(expand("{dir}/{subdir}",
+                dir=[config["data_dir"],config["tmp_dir"]],
+                subdir=config["subdirs"]["assembly"]))
+            shell("{config[bin_dir]}/assembly.py assemble_spades {input.taxfilt_reads} {output.contigs_spades}")
+            shell("cat {input.contigs_trinity} {output.contigs_spades} > {output.contigs_trinity_spades}")
```

Comment:
Why are you `cat`ing the output instead of using `--contigsUntrusted` to feed it into SPAdes?